x/gateway/keeper/msg_server_update_param.go

```diff
@@ -3,15 +3,45 @@ package keeper
 import (
 	"context"
 
-	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/pokt-network/poktroll/x/gateway/types"
 )
 
-func (k msgServer) UpdateParam(goCtx context.Context, msg *types.MsgUpdateParam) (*types.MsgUpdateParamResponse, error) {
-	ctx := sdk.UnwrapSDKContext(goCtx)
+func (k msgServer) UpdateParam(ctx context.Context, msg *types.MsgUpdateParam) (*types.MsgUpdateParamResponse, error) {
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
 
-	// TODO: Handling the message
-	_ = ctx
+	if k.GetAuthority() != msg.Authority {
+		return nil, types.ErrGatewayInvalidSigner.Wrapf("invalid authority; expected %s, got %s", k.GetAuthority(), msg.Authority)
+	}
 
-	return &types.MsgUpdateParamResponse{}, nil
+	params := k.GetParams(ctx)
+
+	switch msg.Name {
+	case types.ParamMinStake:
+		value, ok := msg.AsType.(*types.MsgUpdateParam_AsCoin)
+		if !ok {
+			return nil, types.ErrGatewayParamInvalid.Wrapf("unsupported value type for %s param: %T", msg.Name, msg.AsType)
+		}
+
+		params.MinStake = value.AsCoin
+	default:
+		return nil, types.ErrGatewayParamInvalid.Wrapf("unsupported param %q", msg.Name)
+	}
+
+	// Perform a global validation on all params, which includes the updated param.
+	// This is needed to ensure that the updated param is valid in the context of all other params.
+	if err := params.Validate(); err != nil {
```

Comment:
```suggestion
	if err := params.ValidateBasic(); err != nil {
```

Reply:
I chose not to rename it this time. My rationale is that it requires changing it in lots of other places which is not conducive to making the "adding parameters" step-by-step instructions any more concise. :sweat_smile: 

Do you have a strong opinion on this and/or a rationale that you could share?

Reply:
Mostly because we're doing it everywhere else :smile: